test/iterator-test.js

```diff
@@ -35,35 +42,280 @@ make('iterator is seekable', function (db, t, done) {
   })
 })
 
-make('reverse seek in the middle', function (db, t, done) {
+make('iterator is seekable with buffer', function (db, t, done) {
+  var ite = db.iterator()
+  ite.seek(new Buffer('two'))
+  ite.next(function (err, key, value) {
+    t.error(err, 'no error from next()')
+    t.same(key.toString(), 'two', 'key matches')
```

Comment:
`t.same` uses `==` whereas `t.equal` uses `===`. i think we should use `t.equal` where possible